lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -349,6 +356,11 @@ def attribute_type
           end
         end
 
+        def model_reflect_on_associations?(associations)
```

Comment:
Maybe this ought to be called `model_has_associations?` ?

And instead of getting all given associations and testing that it's not empty after compacting, why not say:

``` ruby
associations.any? { |association| !!model.reflect_on_association(association) }
```